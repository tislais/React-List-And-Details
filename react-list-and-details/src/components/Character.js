import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ name, affiliation, allies, enemies, photoUrl }) => (
  <figure>
    <img src={photoUrl} alt={name} />
    <figcaption>
      <p>{name}</p>
      <p>{affiliation}</p>
      <p>{allies}</p>
      <p>{enemies}</p>
      <p>{photoUrl}</p>
    </figcaption>
  </figure>
);

Character.propTypes = {
  name: PropTypes.string.isRequired,
  affiliation: PropTypes.string.isRequired,
  allies: PropTypes.string.isRequired,
  enemies: PropTypes.string.isRequired,
  photoUrl: PropTypes.string.isRequired,
};

export default Character;