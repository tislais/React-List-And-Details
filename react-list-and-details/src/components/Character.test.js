import { render, screen } from '@testing-library/react';
import Character from './Character';

test('renders learn react link', () => {
  render(<Character />);
  const linkElement = screen.getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});
