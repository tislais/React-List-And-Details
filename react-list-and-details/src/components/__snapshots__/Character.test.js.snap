// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<DocumentFragment>
  <figure>
    <img
      alt="46th Earth King"
      src="https://vignette.wikia.nocookie.net/avatar/images/5/51/46th_Earth_King.png/revision/latest?cb=20130627160441"
    />
    <figcaption>
      <p>
        46th Earth King
      </p>
      <p />
      <p>
        Royal Earthbender Guards
      </p>
      <p>
        Chin
      </p>
    </figcaption>
  </figure>
</DocumentFragment>
`;
